test/e2e/packagemanifest_e2e_test.go

```diff
@@ -3,170 +3,196 @@ package e2e
+
+				_, err = fetchCatalogSource(crc, catalogSourceName, testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("querying the package manifest using the package name", func() {
+
+					expectedStatus := packagev1.PackageManifestStatus{
+						CatalogSource:          catalogSourceName,
+						CatalogSourceNamespace: testNamespace,
+						PackageName:            packageName,
+						Channels: []packagev1.PackageChannel{
+							{
+								Name:           stableChannel,
+								CurrentCSV:     packageStable,
+								CurrentCSVDesc: packagev1.CreateCSVDescription(&csv, string(csvJSON)),
+							},
+						},
+						DefaultChannel: stableChannel,
+					}
+
+					pm, err := fetchPackageManifest(pmc, testNamespace, packageName, packageManifestHasStatus)
+					Expect(err).ToNot(HaveOccurred(), "error getting package manifest")
+					Expect(pm).ShouldNot(BeNil())
+
+					Expect(pm.GetName()).Should(Equal(packageName))
+					Expect(pm.Status).Should(Equal(expectedStatus))
+					Expect(pm.Status.Channels[0].CurrentCSVDesc.MinKubeVersion).Should(Equal("0.0.0"))
+					Expect(pm.Status.Channels[0].CurrentCSVDesc.RelatedImages[0]).Should(Equal(*dummyImage))
+					Expect(pm.Status.Channels[0].CurrentCSVDesc.NativeAPIs).Should(Equal(csv.Spec.NativeAPIs))
+					Expect(pm.GetLabels()["projected"]).Should(Equal("label"))
+					Expect(pm.GetLabels()["operatorframework.io/arch.amd64"]).Should(Equal("supported"))
+					Expect(pm.GetLabels()["operatorframework.io/os.linux"]).Should(Equal("supported"))
+
+					// Get a PackageManifestList and ensure it has the correct items
+					pmList, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
+					Expect(err).NotTo(HaveOccurred(), "could not access package manifests list meta")
+					Expect(pmList.ListMeta).ShouldNot(BeNil(), "package manifest list metadata empty")
+					Expect(pmList.Items).ShouldNot(BeNil())
+				})
+
+			})
 		})
-		require.NoError(GinkgoT(), err, "package-server not available")
-
-		_, err = fetchCatalogSource(GinkgoT(), crc, source.GetName(), testNamespace, catalogSourceRegistryPodSynced)
-		require.NoError(GinkgoT(), err)
-
-		pm, err := fetchPackageManifest(GinkgoT(), pmc, testNamespace, packageName, packageManifestHasStatus)
-		require.NoError(GinkgoT(), err, "error getting package manifest")
-		require.NotNil(GinkgoT(), pm)
-		require.Equal(GinkgoT(), packageName, pm.GetName())
-
-		relatedImages := pm.Status.Channels[0].CurrentCSVDesc.RelatedImages
-		require.Equal(GinkgoT(), len(expectedRelatedImages), len(relatedImages))
+	})
 
-		for _, v := range relatedImages {
-			_, ok := expectedRelatedImages[v]
-			require.True(GinkgoT(), ok, "Expect this image %s to exist in the related images list\n", v)
-		}
+	Context("Given an operator-registry image", func() {
+		Context("When a CatalogSource is created using gRPC catalog source type", func() {
+			When("Package manifest is available", func() {
```

Comment:
I wonder if it would be clearer to change the test from:

1. poll until list package manifests returns without error
2. require that get catalog source returns without error
3. poll until get package manifest returns without error

to:

1. poll until get package manifest returns without error

It seems to me that we're checking for multiple related conditions, when a failure in any one of them would imply a failure of the primary test condition.